internal/util/util.go

```diff
@@ -615,3 +616,41 @@ func HttpRequest(method string, url string, headers []string) (resp *http.Respon
 	}
 	return client.Do(req)
 }
+
+// SafeWrite performs the following actions:
+// 1. If not exists try to write the file and return
+// 2. Create a derived temporary file first
+// 2.1. If this fails remove the corrupted temporary file and return with error
+// 3. Rename the temporary to the target file or overwrite the target file
+// 3.1. If this fails remove the temporary in case of rename otherwise keep the file and return with error
+// 4. Remove the temporary file
```

Comment:
This description basically just repeats what is more or less self-evident from the code.

Maybe it's good that the description is so precise, on the other hand it doesn't even try to explain the high-level purpose of all this, doesn't mention that `rename` provides atomic write, doesn't explain why we don't always want atomic write, etc...

Reply:
> This description basically just repeats what is more or less self-evident from the code.

This is a compliment to your code, BTW.

Reply:
Ok, so we should most probably drop the self-evident documentation and precise the purpose (for `SafeWrite` & `safeWrite`):
It shall provide atomic writes and keep backups of the files where the atomic write didn't succeed and so on...

Reply:
I think we shouldn't override the usual meaning of the word "atomic". Only `rename=true` writes are atomic (i.e. guarantee that the file is never left in a corrupted state), `rename=false` writes are not atomic, by definition (but still are safe, or at least are supposed to be, since they allow recovering a corrupted file afterwards).

(This BTW also means that the name of this PR is misleading too, but that's not so important.)